datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -2523,3 +2525,75 @@ fn roundtrip_window() {
     roundtrip_expr_test(test_expr6, ctx.clone());
     roundtrip_expr_test(text_expr7, ctx);
 }
+
+#[tokio::test]
+async fn roundtrip_recursive_query() {
+    #[derive(Debug)]
+    pub struct EmptyTableCodec;
+
+    impl LogicalExtensionCodec for EmptyTableCodec {
+        fn try_decode(
+            &self,
+            _buf: &[u8],
+            _inputs: &[LogicalPlan],
+            _ctx: &SessionContext,
+        ) -> Result<Extension, DataFusionError> {
+            not_impl_err!("No extension codec provided")
+        }
+
+        fn try_encode(
+            &self,
+            _node: &Extension,
+            _buf: &mut Vec<u8>,
+        ) -> Result<(), DataFusionError> {
+            not_impl_err!("No extension codec provided")
+        }
+
+        fn try_decode_table_provider(
+            &self,
+            _buf: &[u8],
+            _table_ref: &TableReference,
+            schema: SchemaRef,
+            _ctx: &SessionContext,
+        ) -> Result<Arc<dyn TableProvider>, DataFusionError> {
+            let table = MemTable::try_new(schema, vec![vec![]])?;
+            Ok(Arc::new(table))
+        }
+
+        fn try_encode_table_provider(
+            &self,
+            _table_ref: &TableReference,
+            _node: Arc<dyn TableProvider>,
+            _buf: &mut Vec<u8>,
+        ) -> Result<(), DataFusionError> {
+            Ok(())
```

Comment:
It seems that we need to [encode](https://github.com/apache/datafusion/blob/c48f12d27cc60ae5e9c3d0791d553453f0dd7001/datafusion/proto/src/logical_plan/mod.rs#L964) `CteWorkTable` and then recreate a `CteWorkTable` when decoding.🤔